airflow/jobs/local_task_job.py

```diff
@@ -80,14 +80,15 @@ def signal_handler(signum, frame):
             self.log.error("Received SIGTERM. Terminating subprocesses")
             self.on_kill()
             self.task_instance.refresh_from_db()
-            if self.task_instance.state not in State.finished:
-                self.task_instance.set_state(State.FAILED)
-            self.task_instance._run_finished_callback(  # pylint: disable=protected-access
-                error="task received sigterm"
-            )
+            if self.task_instance.state == State.RUNNING:
+                # This only happens when a pod is deleted
+                # killing the task_runner by sending sigterm.
+                # TI sigterm is handled at taskinstance.py
+                self.handle_task_exit(128 + signum)
```

Comment:
@ashb The test I had previously for sigterm was not hitting here. I'm only able to hit here when I delete a running kubernetes pod. I wonder how we can test this?